### PR TITLE
tests: kernel: submit a simple test for zero-latency irqs

### LINKED ISSUE #16901: No test coverage for CONFIG_ZERO_LATENCY_IRQS
This ARM feature exists and appears correct (and builds, at least), but is exercised anywhere in the existing test suite.  It needs validation if we're going to claim it as a feature.

### PR DESCRIPTION
**Pull request by ioannisg**
Fixes #16901 

The commit contributes a simple test for the Zero-Latency
IRQ feature (CONFIG_ZERO_LATENCY_IRQS=y) for ARM platforms.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

File: tests/kernel/arm_zero_latency_irqs/src/main.c [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (nordic-krch) [inline tests/kernel/arm_zero_latency_irqs/src/main.c]: why is it split into two files?

File: tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+void test_arm_zero_latency_irqs(void)
+{
+	/* Determine an NVIC IRQ line that is not currently in use. */
+	int i, key;
+
+	zassert_false(test_flag, "Test flag not initialized to zero\n");
+
+	for (i = CONFIG_NUM_IRQS - 1; i >= 0; i--) {
+
```
Comment (nordic-krch) [inline tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c]: new line seems redundant

File: tests/kernel/arm_zero_latency_irqs/src/main.c (hunk 1 above)
Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/src/main.c]: I suppose, I just followed the arm_irq_vector_table test structure

File: tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c (hunk 2 above)
Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c]: Yeap, removing it now.

**Comment by ioannisg** [comment]
@cvinayak could you also take a quick look?

File: tests/kernel/arm_zero_latency_irqs/README.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    make run
+
+---------------------------------------------------------------------------
+
+Troubleshooting:
+
+Problems caused by out-dated project information can be addressed by
+issuing one of the following commands then rebuilding the project:
+
+    make clean          # discard results of previous builds
```
Comment (carlescufi) [inline tests/kernel/arm_zero_latency_irqs/README.txt]: Let's use `ninja/make clean` and `ninja/make pristine`

**Review by cvinayak — APPROVED** [review]
LGTM

File: tests/kernel/arm_zero_latency_irqs/README.txt (hunk 3 above)
Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/README.txt]: Will update this, tnem

Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/README.txt]: Fixed.

File: tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+void test_arm_zero_latency_irqs(void)
+{
+	/* Determine an NVIC IRQ line that is not currently in use. */
+	int i, key;
+
+	zassert_false(test_flag, "Test flag not initialized to zero\n");
+
+	for (i = CONFIG_NUM_IRQS - 1; i >= 0; i--) {
+		if (NVIC_GetEnableIRQ(i) == 0) {
```
Comment (carlescufi) [inline tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c]: Can you add a comment as to why `GetEnableIRQ()` allows you to find an available interrupt?

Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c]: Something like what we have in line 22?

Comment (ioannisg) [reply tests/kernel/arm_zero_latency_irqs/src/arm_zero_latency_irqs.c]: Or a more detailed comment, e.g.
 `/*
   *Interrupts configured statically with IRQ_CONNEC(T..) are automatically enabled. NVIC_GetEnableIRQ() returning
   *false implies that the IRQ line is not currently in use by Zephyr.
  */`

**Review by andyross — APPROVED** [review]
Looks great to me

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**